File: server/s2game/src/main/java/com/spectopics/s2game/dto/serverPayloads/LobbyJoinedPayload.java
package com.spectopics.s2game.dto.serverPayloads;

import com.spectopics.s2game.models.LobbyState;

public class LobbyJoinedPayload {
    // Tells the clients that someone has successfully joined the lobby, and gives them the current state of the lobby.
    public LobbyState lobbyState;
    public String playerId;

    public LobbyJoinedPayload(LobbyState lobbyState, String playerId) {
        this.lobbyState = lobbyState;
        this.playerId = playerId;
    }
}

File: front-end/src/dto/serverPayloads/LobbyJoinedPayload.js
import { LobbyState } from '../../models/LobbyState.js';

export class LobbyJoinedPayload {
    constructor(lobbyState, playerId) {
        this.lobbyState = lobbyState;
        this.playerId = playerId;
    }

    static fromDict(data) {
        return new LobbyJoinedPayload(
            LobbyState.fromDict(data.lobbyState),
            data.playerId
        );
    }
}
